RichTextData: use a private property

jsnajdr opened the pull request:
In `RichTextData`, use a private class instance property for the private data, instead of a symbol-indexed public property. Private properties are now standard, they should work.

jsnajdr committed:
RichTextData: use a private property

jsnajdr commented on packages/rich-text/src/create.js:
```diff
@@ -138,22 +137,15 @@ export class RichTextData {
-			this.originalHTML ||
-			toHTMLString( { value: this[ RichTextInternalData ] } )
-		);
+		return this.originalHTML || toHTMLString( { value: this.#value } );
```
The way how `originalHTML` is defined in `fromHTMLElement` is not entirely correct. It effectively creates a _live binding_ between the `htmlElement` and the `RichTextData` value. If the element's contents change, then `toHTMLString()` will read the latest `.innerHTML` value in the getter.

Other parts of the `RichTextData` value are however not live: the `toPlainText` method, the `.text` getter, etc. The object's content can be inconsistent and out of sync.

ellatrix replied:
RichTextData should not be mutable

jsnajdr replied:
Then we need to read `htmlElement.innerHTML` only once in `RichTextData` constructor, not on every call of `.toHTMLString`.

ellatrix replied:
We do it only once in the static method? `originalHTML` is a value not a getter?

ellatrix replied:
Or do you mean `toHTMLString()`? We shouldn't calculate it in the constructor for performance, we sure we could cache it here for the next call

ellatrix replied:
(We only want to serialise when it's needed)

jsnajdr replied:
> `originalHTML` is a value not a getter?

Oh, somehow I overlooked this 🙂 It's alright then.

ellatrix commented:
I have tried this but it didn't work for me. Would be cool if it works :)

jsnajdr commented:
Oh, TypeScript makes a complaint I don't understand:
```
error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.
```
because we define `target: 'esnext'` in our `tsconfig.json`.

jsnajdr committed:
check-build-type-declaration-files: pass target and moduleResolution options to tsc

ellatrix reviewed (approved):
Very nice :)

jsnajdr commented:
I fixed the TypeScript error. It was the `check-build-type-declaration-files.js` script that runs `tsc` on individual files. It doesn't build/check a project that has a `tsconfig.json`, but individual files, and no config is loaded. I added the required `target` and `moduleResolution` options on command line.

Maybe @noahtallen or @sirreal know a way how to make `tsc` read a `tsconfig.json`.

sirreal commented:
The goal is to run `tsc` on arbitrary files but with a provided compiler options from a tsconfig file? I believe `--project` and `--build` are the options to pass a tsconfig, but those will typically determine whats included in a compilation as well. I don't know whether you can use those options with a source file to process argument.

jsnajdr commented:
> The goal is to run `tsc` on arbitrary files but with a provided compiler options from a tsconfig file?

Yes, the goal is to typecheck one specific file with a config for a particular project.

borislavfil commented:
Hello, I am getting "SyntaxError: Invalid character: '#'" on Safari IOS 14.3 and I have narrowed it down to this MR. It seems to work correctly on any other browser I have tested (Windows and Android). I am not sure if I should create a new Issue
